cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -140,10 +140,15 @@ func (runtime *CRIRuntime) RemoveContainers(containers []string) error {
 func (runtime *DockerRuntime) RemoveContainers(containers []string) error {
 	errs := []error{}
 	for _, container := range containers {
-		out, err := runtime.exec.Command("docker", "rm", "--force", "--volumes", container).CombinedOutput()
+		out, err := runtime.exec.Command("docker", "stop", container).CombinedOutput()
 		if err != nil {
 			// don't stop on errors, try to remove as many containers as possible
-			errs = append(errs, errors.Wrapf(err, "failed to remove running container %s: output: %s, error", container, string(out)))
+			errs = append(errs, errors.Wrapf(err, "failed to stop running container %s: output: %s, error", container, string(out)))
+		} else {
+			out, err = runtime.exec.Command("docker", "rm", "--volumes", container).CombinedOutput()
```

Comment:
the question here is, should we remove a container even if stop failed?

Reply:
My point is that ContainerRuntime behaves in the same way. DockerRuntime should return error on container stopping failures just like CRIRuntime.